### ISSUE TITLE
dimesnions in train loop X vs X,1 or 1, X

### ISSUE THREAD

**Issue by Hitomamacs** [ZIGTinyBook/TheTinyBook#50]

**Comment by Hitomamacs** [comment]
look at train with dataloader training loop in compute loss and accuracy i did strange magic
